### ISSUE TITLE
vllm 0.2.7 with tensor-parallel > 1 inference mode in place error 

### ISSUE THREAD

**Issue by lroberts7** [vllm-project/vllm#2620]
In trying to debug tensor parallel on 0.2.7 and have exhausted possible ideas. 
Trying to run mixtral 8X7b model which requires 2 gpu devices I have 8X 80GB VRAM A100 cuda machine. 

```bash
lroberts@GPU77B9:~/llm_quantization$ FLASK_APP=quantized_flask_app.py FLASK_ENV=debug python3.10 -m flask run 
 * Serving Flask app 'quantized_flask_app.py' (lazy loading)
 * Environment: debug
 * Debug mode: off
/usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
16384
INFO 2024-01-26 22:13:10,261 abc_etal.py:195 unknown_model_name:unknown_model_version
                             Hello! logging initialized, starting up... 
INFO 2024-01-26 22:13:10,261 abc_etal.py:196 unknown_model_name:unknown_model_version
                             Git commit of model: unknown_git_commit 
INFO 2024-01-26 22:13:10,261 abc_etal.py:197 unknown_model_name:unknown_model_version
                             Git commit of cuda torch base: unknown_git_commit 
INFO 2024-01-26 22:13:11,883 abc_etal.py:200 unknown_model_name:unknown_model_version
                             Compute device available: cuda 
WARNING 01-26 22:13:14 config.py:506] Casting torch.bfloat16 to torch.float16.
WARNING 01-26 22:13:14 config.py:176] awq quantization is not fully optimized yet. The speed can be slower than non-quantized models.
2024-01-26 22:13:17,748 INFO worker.py:1673 -- Started a local Ray instance.
INFO 01-26 22:13:19 llm_engine.py:72] Initializing an LLM engine with config: model='/home/lroberts/NexusRaven-13B-AWQ/', tokenizer='/home/lroberts/NexusRaven-13B-AWQ/presaved_tokenizer', tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.float16, max_seq_len=16384, download_dir=None, load_format=auto, tensor_parallel_size=2, quantization=awq, enforce_eager=False, seed=0)
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/__main__.py", line 3, in <module>
    main()
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 990, in main
    cli.main(args=sys.argv[1:])
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 596, in main
    return super().main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 845, in run_command
    app = DispatchingApp(info.load_app, use_eager_loading=eager_loading)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 321, in __init__
    self._load_unlocked()
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 346, in _load_unlocked
    self._app = rv = self.loader()
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 402, in load_app
    app = locate_app(self, import_name, name)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 273, in locate_app
    return find_best_app(script_info, module)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 68, in find_best_app
    app = call_factory(script_info, app_factory)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 119, in call_factory
    return app_factory(*args, **kwargs)
  File "/home/lroberts/llm_quantization/quantized_flask_app.py", line 244, in create_app
    app.model = ChatEn()
  File "/home/lroberts/llm_quantization/abc_etal.py", line 219, in __init__
    self.model = LLM(
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/entrypoints/llm.py", line 106, in __init__
    self.llm_engine = LLMEngine.from_engine_args(engine_args)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 350, in from_engine_args
    engine = cls(*engine_configs,
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 112, in __init__
    self._init_cache()
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 303, in _init_cache
    num_blocks = self._run_workers(
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 977, in _run_workers
    driver_worker_output = getattr(self.driver_worker,
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/worker.py", line 111, in profile_num_available_blocks
    self.model_runner.profile_run()
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/model_runner.py", line 592, in profile_run
    self.execute_model(seqs, kv_caches)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/model_runner.py", line 516, in execute_model
    self.prepare_input_tensors(seq_group_metadata_list))
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/model_runner.py", line 480, in prepare_input_tensors
    broadcast_tensor_dict(metadata_dict, src=0)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/model_executor/parallel_utils/communication_op.py", line 158, in broadcast_tensor_dict
    torch.distributed.broadcast_object_list([metadata_list],
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/c10d_logger.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 2603, in broadcast_object_list
    broadcast(object_sizes_tensor, src=src, group=group)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/c10d_logger.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 1906, in broadcast
    work = default_pg.broadcast([tensor], opts)
RuntimeError: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.
```

error message always the same as above (and same stack trace).

what I've tried

```bash
pip uninstall torch  -y 
pip install --upgrade torch --index-url https://download.pytorch.org/whl/cu118

pip uninstall xformers -y 
pip install --upgrade xformers --index-url https://download.pytorch.org/whl/cu118
```

also tried `cu121` build for xformers and torch which results in the same error. 

I've also seen the error inference mode error on vllm==0.2.3.post1 release 

Nividia details: 
```bash
lroberts@GPU77B9:~/llm_quantization$ nvidia-smi 
Fri Jan 26 22:18:56 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                  Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf            Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-80GB           On | 00000000:07:00.0 Off |                    0 |
| N/A   34C    P0               68W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB           On | 00000000:0A:00.0 Off |                    0 |
| N/A   31C    P0               65W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   2  NVIDIA A100-SXM4-80GB           On | 00000000:47:00.0 Off |                    0 |
| N/A   32C    P0               63W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   3  NVIDIA A100-SXM4-80GB           On | 00000000:4D:00.0 Off |                    0 |
| N/A   35C    P0               69W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   4  NVIDIA A100-SXM4-80GB           On | 00000000:87:00.0 Off |                    0 |
| N/A   36C    P0               68W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   5  NVIDIA A100-SXM4-80GB           On | 00000000:8D:00.0 Off |                    0 |
| N/A   33C    P0               69W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   6  NVIDIA A100-SXM4-80GB           On | 00000000:C7:00.0 Off |                    0 |
| N/A   32C    P0               66W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   7  NVIDIA A100-SXM4-80GB           On | 00000000:CA:00.0 Off |                    0 |
| N/A   34C    P0               67W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
```

also I confirmed I still see the issue on a development version build off commit 
https://github.com/vllm-project/vllm/commit/5265631d15d59735152c8b72b38d960110987f10

from 14 hours ago. The build will work for development version on `main` and I can use with `tp=1` but mixtral needs tensor-parallel equal 2 at minimum. 
s 
During the setup of tensor-parallel=2 I ran into pydantic issue noted here: 

https://github.com/vllm-project/vllm/issues/2619

using pip to uninstall pydantic version 2 and force install of version 1.10 
```bash
pip uninstall pydantic -y 
pip install pydantic==1.10
```
then results in the same issue indicated above (with `--tensor-parallel 2`): 

```
...
RuntimeError: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.
```

my question is how to get mixtral model to serve inferences?

 I am using https://huggingface.co/mistralai/Mixtral-8x7B-Instruct-v0.1 model direct from HF

**Comment by lroberts7** [comment]
I followed the changes merged over the weekend. I was hoping that the latest change to the kvcache config would fix my issue, it seems no, because I did

```bash 
git clean -xdf # removes __pycache__ files
# note I'm on tip of main branch 
git log -n 2
commit b72af8f1eded6f5838be29eb6093ab0e0e0c240c (HEAD -> main, origin/main, origin/HEAD)
Author: zhaoyang-star <zhaoyangstar@foxmail.com>
Date:   Mon Jan 29 14:47:39 2024 +0800

    Fix error when tp > 1 (#2644)
    
    Co-authored-by: zhaoyang-star <zhao.yang16@zte.com.cn>

commit 9090bf02e74334a8020b454814e0d00fa780fd79
Author: zhaoyang-star <zhaoyangstar@foxmail.com>
Date:   Mon Jan 29 08:43:54 2024 +0800

    Support FP8-E5M2 KV Cache (#2279)
    
    Co-authored-by: zhaoyang <zhao.yang16@zte.com.cn>
    Co-authored-by: Zhuohan Li <zhuohan123@gmail.com>

pip install -e . # rebuild vllm 
```
after ~10 minutes to build I retry and still the same error: 

```
FLASK_APP=quantized_flask_app.py FLASK_ENV=debug python3.10 -m flask run
 * Serving Flask app 'quantized_flask_app.py' (lazy loading)
 * Environment: debug
 * Debug mode: off

(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/__main__.py", line 3, in <module>
    main()
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 990, in main
    cli.main(args=sys.argv[1:])
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 596, in main
    return super().main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 845, in run_command
    app = DispatchingApp(info.load_app, use_eager_loading=eager_loading)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 321, in __init__
    self._load_unlocked()
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 346, in _load_unlocked
    self._app = rv = self.loader()
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 402, in load_app
    app = locate_app(self, import_name, name)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 273, in locate_app
    return find_best_app(script_info, module)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 68, in find_best_app
    app = call_factory(script_info, app_factory)
  File "/home/lroberts/.local/lib/python3.10/site-packages/flask/cli.py", line 119, in call_factory
    return app_factory(*args, **kwargs)
  File "/home/lroberts/llm_quantization/quantized_flask_app.py", line 244, in create_app
    app.model = ChatEn()
  File "/home/lroberts/llm_quantization/abc_etal.py", line 219, in __init__
    self.model = LLM(
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/entrypoints/llm.py", line 109, in __init__
    self.llm_engine = LLMEngine.from_engine_args(engine_args)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 356, in from_engine_args
    engine = cls(*engine_configs,
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 114, in __init__
    self._init_cache()
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 308, in _init_cache
    num_blocks = self._run_workers(
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/engine/llm_engine.py", line 983, in _run_workers
    driver_worker_output = getattr(self.driver_worker,
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/worker.py", line 116, in profile_num_available_blocks
    self.model_runner.profile_run()
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/model_runner.py", line 599, in profile_run
    self.execute_model(seqs, kv_caches)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/model_runner.py", line 523, in execute_model
    self.prepare_input_tensors(seq_group_metadata_list))
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/worker/model_runner.py", line 486, in prepare_input_tensors
    broadcast_tensor_dict(metadata_dict, src=0)
  File "/home/lroberts/update-vllm-env/vllm-source/vllm/vllm/model_executor/parallel_utils/communication_op.py", line 168, in broadcast_tensor_dict
    torch.distributed.broadcast_object_list([metadata_list],
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/c10d_logger.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 2603, in broadcast_object_list
    broadcast(object_sizes_tensor, src=src, group=group)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/c10d_logger.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/lroberts/.local/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 1906, in broadcast
    work = default_pg.broadcast([tensor], opts)
RuntimeError: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
[W CudaIPCTypes.cpp:15] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]

```

So I'm still unable to serve inferences on `-tp >1` case.

**Comment by lroberts7** [comment]
@simon-mo I think this is now resolved for my dev build. 

What I did to confirm:

`git clean -xdf`

install editable build via pip install instructions: 
`pip install -e . --no-build-isolation`

then confirm it's using the editable build:

`pip list | grep vllm`

which gives me 
```
lroberts@GPU77B9:~/llm_quantization$ pip list | grep vllm 
vllm                      0.2.7+cu120      /home/lroberts/update-vllm-env/vllm-source/vllm
```

now stand up server with the mixtral model: 

```bash
lroberts@GPU77B9:~/llm_quantization$ lroberts@GPU77B9:~/llm_quantization$ FLASK_APP=quantized_flask_app.py FLASK_ENV=debug python3.10 -m flask run  
 * Serving Flask app 'quantized_flask_app.py' (lazy loading)
 * Environment: debug
 * Debug mode: off
/usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
32768
INFO 2024-01-29 23:18:38,674 abc_etal.py:197 unknown_model_name:unknown_model_version
                             Hello! logging initialized, starting up... 
INFO 2024-01-29 23:18:38,674 abc_etal.py:198 unknown_model_name:unknown_model_version
                             Git commit of model: unknown_git_commit 
INFO 2024-01-29 23:18:38,674 abc_etal.py:199 unknown_model_name:unknown_model_version
                             Git commit of cuda torch base: unknown_git_commit 
INFO 2024-01-29 23:18:40,270 abc_etal.py:202 unknown_model_name:unknown_model_version
                             Compute device available: cuda 
2024-01-29 23:18:44,368 INFO worker.py:1724 -- Started a local Ray instance.
INFO 01-29 23:18:45 llm_engine.py:72] Initializing an LLM engine with config: model='/home/lroberts/Mixtral-8x7B-Instruct-v0.1/', tokenizer='/home/lroberts/Mixtral-8x7B-Instruct-v0.1/presaved_tokenizer', tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=32768, download_dir=None, load_format=auto, tensor_parallel_size=2, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto, seed=0)
(raylet) /usr/lib/python3/dist-packages/requests/__init__.py:87: RequestsDependencyWarning: urllib3 (2.1.0) or chardet (5.2.0) doesn't match a supported version!
(raylet)   warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
INFO 01-29 23:19:13 llm_engine.py:322] # GPU blocks: 20934, # CPU blocks: 4096
INFO 01-29 23:19:16 model_runner.py:632] Capturing the model for CUDA graphs. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI.
INFO 01-29 23:19:16 model_runner.py:636] CUDA graphs can take additional 1~3 GiB memory per GPU. If you are running out of memory, consider decreasing `gpu_memory_utilization` or enforcing eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.
(RayWorkerVllm pid=3968880) INFO 01-29 23:19:16 model_runner.py:632] Capturing the model for CUDA graphs. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI.
(RayWorkerVllm pid=3968880) INFO 01-29 23:19:16 model_runner.py:636] CUDA graphs can take additional 1~3 GiB memory per GPU. If you are running out of memory, consider decreasing `gpu_memory_utilization` or enforcing eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.
INFO 01-29 23:19:23 custom_all_reduce.py:195] Registering 2275 cuda graph addresses
INFO 01-29 23:19:23 model_runner.py:698] Graph capturing finished in 7 secs.
INFO 2024-01-29 23:19:23,606 abc_etal.py:232 unknown_model_name:unknown_model_version
                             Startup completed! 
INFO 2024-01-29 23:19:23,609 _internal.py:224 unknown_model_name:unknown_model_version
                             WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000 
INFO 2024-01-29 23:19:23,609 _internal.py:224 unknown_model_name:unknown_model_version
                             Press CTRL+C to quit 
(RayWorkerVllm pid=3968880) INFO 01-29 23:19:23 custom_all_reduce.py:195] Registering 2275 cuda graph addresses
(RayWorkerVllm pid=3968880) INFO 01-29 23:19:23 model_runner.py:698] Graph capturing finished in 7 secs.
[OpenAIMessage(role='system', content='You are a helpful assistant.'), OpenAIMessage(role='user', content='Tell me a few reasons why someone might consider higher education. Do not repeat yourself. Response:  ')]
32768
Processed prompts: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:13<00:00, 13.80s/it]

```

issue simple curl request:

```bash
curl -v --trace-time -X POST -H "Content-Type: application/json" --data '{"max_tokens": 500, "messages": [{"content": "You are a helpful assistant.","role": "system"}, {"content": "Tell me a few reasons why someone might consider higher education. Do not repeat yourself. Response:  ","role": "user"}], "model": "gpt-3.5-turbo", "temperature": 0}' http://localhost:5000/sequence-generation/chat/json  
```

get back output 

```
{"choices":[{"finish_reason":"stop","index":0,"message":{"content":" 1. Career Advancement: Higher education can lead to better job opportunities and higher salaries. Many professions require a degree as a basic prerequisite.\n\n2. Expanded Knowledge: Higher education allows students to delve deeper into a specific field, gaining a more comprehensive understanding of the subject matter.\n\n3. Critical Thinking Skills: Higher education often encourages students to think critically, question, and analyze information, which are valuable skills in any career.\n\n4. Networking Opportunities: Universities and colleges provide numerous opportunities to meet and connect with professionals, professors, and other students, which can be beneficial for future career prospects.\n\n5. Personal Development: Higher education can help students develop personally, fostering independence, time management skills, and self-confidence.\n\n6. Lifelong Learning: Higher education promotes a culture of lifelong learning, encouraging students to continue learning and growing beyond their formal education.\n\n7. Financial Security: Higher education can lead to more stable and secure financial futures, with graduates often earning significantly more than those with only a high school diploma.\n\n8. Intellectual Challenge: Higher education offers the opportunity to engage in complex and challenging academic work, stimulating the mind and fostering intellectual growth.","role":"assistant"}}],"created":1706570672,"id":"llama-2-7b-chat-hf","object":"chat.completion","usage":{"completion_tokens":285,"prompt_tokens":46,"total_tokens":331}}
```

Thanks! This can be closed. Let me know if you need more specs on the machine or environment if there are issues to triangulate.
